src/Microsoft.OData.Client/Serialization/Serializer.cs

```diff
@@ -405,6 +610,132 @@ internal void WriteNestedResourceInfo(EntityDescriptor entityDescriptor, IEnumer
             }
         }
 
+        /// <summary>
+        /// Writes a nested resource
+        /// </summary>
+        /// <param name="entityDescriptor">The entity</param>
+        /// <param name="relatedLinks">The links related to the entity</param>
+        /// <param name="odataWriter">The ODataWriter used to write the navigation link.</param>
+        internal void WriteDeltaNestedResourceInfo(EntityDescriptor entityDescriptor, IEnumerable<Descriptor> relatedDescriptors, ODataWriterWrapper odataWriter)
+        {
+            Dictionary<string, List<Descriptor>> groupRelatedLinks = new Dictionary<string, List<Descriptor>>(EqualityComparer<string>.Default);
+            foreach (Descriptor end in relatedDescriptors)
+            {
+                List<Descriptor> relatedLinks = null;
+                switch (end.DescriptorKind)
+                {
+                    case DescriptorKind.Entity:
+                        var entityObj = (EntityDescriptor)end;
+                        var entityName = entityObj.ParentProperty;
+                        if (groupRelatedLinks.TryGetValue(entityName, out relatedLinks))
+                        {
+                            relatedLinks.Add(entityObj);
+                        }
+                        else
+                        {
+                            if (relatedLinks == null)
+                            {
+                                relatedLinks = new List<Descriptor>();
+                            }
+
+                            relatedLinks.Add(entityObj);
+                            groupRelatedLinks.Add(entityName, relatedLinks);
+                        }

```

Comment:
Not sure what this block is doing. Is it grouping descriptors that are elements of the same collection navigation property in the same list?

Reply:
It is grouping elements that are of the same collection navigation property.